src/ExpressionParser.php

```diff
@@ -105,19 +106,21 @@ public function parseExpression($precedence = 0, $allowArrow = false, bool $depr
             } elseif (isset($op['callable'])) {
                 $expr = $op['callable']($this->parser, $expr);
             } else {
-                $expr1 = $this->parseExpression(self::OPERATOR_LEFT === $op['associativity'] ? $op['precedence'] + 1 : $op['precedence'], true);
+                $previous = $this->setDeprecationCheck(true);
+                try {
+                    $expr1 = $this->parseExpression(self::OPERATOR_LEFT === $op['associativity'] ? $op['precedence'] + 1 : $op['precedence'], true);
+                } finally {
+                    $this->setDeprecationCheck($previous);
+                }
                 $class = $op['class'];
                 $expr = new $class($expr, $expr1, $token->getLine());
             }
 
             $expr->setAttribute('operator', 'binary_'.$token->getValue());
 
-            $previousToken = $token;
-            $token = $this->parser->getCurrentToken();
-        }
+            $this->triggerPrecedenceDeprecations($expr, $token);
 
-        if ($deprecationCheck) {
```

Comment:
Not needed as the `setExplicitParentheses()` is set before the check here. The `deprecationCheck` is only useful for unary operators.